sqstaskmaster/tests/test_local.py

```diff
@@ -0,0 +1,248 @@
+
+class MyHandler(MessageHandler):
+    def __init__(
+        self,
+        message,
+        sqs_timeout,
+        alarm_timeout,
+        run_method,
+        hard_timeout=4 * 60 * 60,
+        **kwargs
+    ):
+        super().__init__(message, sqs_timeout, alarm_timeout, hard_timeout)
+        self.kwargs = kwargs
+        self.run_method = run_method
+
+    def running(self):
+        return True
+
+    def run(self):
+        self.run_method(**self.kwargs)
+
+    def notify(self, exception, context=None):
+        pass
+
+
+def example_consumer(url, do_method):
+    """
+    Consumer is typically a runnable main that is deployed to execute a tasks.
+    This method demonstrates how the integration with the message publisher might be tested
+    :param url: the sqs url
+    :param do_method: hook to provide a method that allows message content validation.
+    """
+    tm = TaskManager(url, queue_constructor=LocalQueue)
+
+    for task, kwargs, message in tm.task_generator(sqs_timeout=30, wait_time=1):
+        logger.info("got task %s with %s and %s", task, kwargs, message)
+        if task == TASK_NAME:
+            with MyHandler(
+                message,
+                sqs_timeout=5,
+                alarm_timeout=1,
+                run_method=do_method,
+                hard_timeout=1,
+                **kwargs
+            ) as handler:
+                logger.info("Calling run method")
+                handler.run()
+        elif task == TASK_STOP:
+            # typically a deployed consumer would run indefinitely waiting for work
+            logger.info("Calling Break for STOP task!")
+            break
+        else:
+            logger.info("Got unexpected task name: %s with message %s", task, message)
+
+
+class TestLocalIntegration(unittest.TestCase):
+    def test_producer_consumer(self):
+        messages = [
+            (TASK_NAME, {"some": "value"}),
+            (TASK_NAME, {"some": "other value"}),
+            (TASK_NAME, {"some": "more values"}),
+            (TASK_NAME, {"some": None}),
+            (TASK_STOP, {"more": "nonesense"}),
+        ]
+
+        example_producer("a_particular_queue", messages)
+        received = []
+
+        def receiver(**kwargs):
+            logger.info("receiver method got %s", kwargs)
+            received.append(kwargs)
+
+        example_consumer("a_particular_queue", receiver)
+
+        self.assertListEqual([k for t, k in messages if t == TASK_NAME], received)
+
+    def test_producer_consumer_timeout(self):
+        messages = [(TASK_NAME, {"some": None}), (TASK_STOP, {"more": "nonesense"})]
+
+        example_producer("a_particular_queue", messages)
+        received = []
+
+        def receiver(**kwargs):
+            logger.info("receiver method got %s", kwargs)
+            received.append(kwargs)
+            time.sleep(2)
+
+        example_consumer("a_particular_queue", receiver)
+
+        self.assertListEqual([k for t, k in messages if t == TASK_NAME], received)
```

Comment:
if I am interpreting this correctly, `test_producer_consumer_timeout` generates a producer and a consumer of messages, though in this case the receiving method has a lag built in... but the lag is less than the limit, so the messages are all still received before hitting the limit?
In that case I wonder if it would be important to add a test showing that a longer lag does trigger the timeout limit and cut the list of received kwargs short?

Reply:
Added comments and additional assertion to explain.